File: algo/hash/programmers_hash_3.py
def solution(clothes):
    answer = {}
    for i in clothes:
        # i[1]는 의상의 종류를 가리킴. 
        # 즉, 답안에 의상의 종류가 있으면 +1을 해주고
        if i[1] in answer: answer[i[1]] += 1
        # 값이 없으면, 1을 넣어준다. 
        else: answer[i[1]] = 1

    cnt = 1
    for i in answer.values():
        cnt *= (i+1)
    return cnt - 1

print(solution([["yellowhat", "headgear"], ["bluesunglasses", "eyewear"], ["green_turban", "headgear"]]))

File: algo/hash/programmers_hash_3.js
function solution(clothes) {
  let answer = 1;

  let type = {};
  for (let cloth of clothes) {
    if (type[cloth[1]]) {
      type[cloth[1]]++;
    } else {
      type[cloth[1]] = 1;
    }
  }

  for (let i of Object.keys(type)) {
    answer *= type[i] + 1;
  }

  return answer - 1;
}
